services/catalog/src/simcore_service_catalog/utils/pools.py

```diff
@@ -0,0 +1,17 @@
+from contextlib import contextmanager
+from concurrent.futures import ProcessPoolExecutor
+
+
+# because there is no shared fastapi library, this is a
+# duplicate of servicelib.pools.non_blocking_process_pool_executor
+@contextmanager
+def non_blocking_process_pool_executor(**kwargs) -> ProcessPoolExecutor:
+    """
+    Avoids default context manger behavior which calls
+    shutdown with wait=True an blocks.
+    """
+    pool = ProcessPoolExecutor(**kwargs)  # pylint: disable=consider-using-with
+    try:
+        yield pool
+    finally:
+        pool.shutdown(wait=False)
```

Comment:
I guess the bug affects thisone as well  ;-)  Make sure it is the same as in ``servicelib.pool``

Reply:
🤦 yes